gsplat/rendering.py

```diff
@@ -314,6 +354,137 @@ def rasterization(
         # make it apple-to-apple with Inria's CUDA Backend.
         colors = torch.clamp_min(colors + 0.5, 0.0)
 
+    # If in distributed mode, we need to scatter the GSs to the destination ranks, based
+    # on which cameras they are visible to, which we already figured out in the projection
+    # stage.
+    if distributed:
+        if packed:
```

Comment:
@TarzanZhao This is the sparse all to all logic, which we called packed mode (only visible GSs are returned from the projection function).

I'm not sure if I'm implementing this part in a most efficient way. Would love to get your thoughts

Reply:
1. depth and radii do not have to use functional version of all2all.
2. collected_cnts does not need functional version of all2all. 
3. [cnt.item() for cnt in cnts] will invoke many times of GPU-CPU communication. The total latency will be large. You can change to transfer cnts to cpu by single call. 

In general, this code has more kernels than ours: https://github.com/nyu-systems/Grendel-GS/blob/0ea84e456d58946aa9708e1932d7c3466edd6a98/gaussian_renderer/__init__.py#L176

More kernels will add significant kernel launch overheads, especially when images have small resolutions.